src/strands/types/session.py

```diff
@@ -0,0 +1,118 @@
+"""Data models for session management."""
+
+import base64
+import inspect
+from dataclasses import dataclass, field
+from datetime import datetime, timezone
+from enum import Enum
+from typing import Any, Dict, cast
+from uuid import uuid4
+
+from ..agent.agent import Agent
+from .content import Message
+
+
+class SessionType(str, Enum):
+    """Enumeration of session types."""
+
+    AGENT = "AGENT"
+
+
+def encode_bytes_values(obj: Any) -> Any:
+    """Recursively encode any bytes values in an object to base64.
+
+    Handles dictionaries, lists, and nested structures.
+    """
+    if isinstance(obj, bytes):
+        return {"__bytes_encoded__": True, "data": base64.b64encode(obj).decode()}
+    elif isinstance(obj, dict):
+        return {k: encode_bytes_values(v) for k, v in obj.items()}
+    elif isinstance(obj, list):
+        return [encode_bytes_values(item) for item in obj]
+    else:
+        return obj
+
+
+def decode_bytes_values(obj: Any) -> Any:
+    """Recursively decode any base64-encoded bytes values in an object.
+
+    Handles dictionaries, lists, and nested structures.
+    """
+    if isinstance(obj, dict):
+        if obj.get("__bytes_encoded__") is True and "data" in obj:
+            return base64.b64decode(obj["data"])
+        return {k: decode_bytes_values(v) for k, v in obj.items()}
+    elif isinstance(obj, list):
+        return [decode_bytes_values(item) for item in obj]
+    else:
+        return obj
+
+
+@dataclass
+class SessionMessage:
+    """Message within a SessionAgent."""
+
+    message: Message
+    message_id: str = field(default_factory=lambda: str(uuid4()))
+    created_at: str = field(default_factory=lambda: datetime.now(timezone.utc).isoformat())
+    updated_at: str = field(default_factory=lambda: datetime.now(timezone.utc).isoformat())
+
+    @classmethod
+    def from_message(cls, message: Message) -> "SessionMessage":
+        """Convert from a Message, base64 encoding bytes values."""
+        bytes_encoded_dict = encode_bytes_values(message)
+        return cls(
+            message=bytes_encoded_dict,
+            message_id=str(uuid4()),
+            created_at=datetime.now(timezone.utc).isoformat(),
+            updated_at=datetime.now(timezone.utc).isoformat(),
+        )
+
+    def to_message(self) -> Message:
+        """Convert SessionMessage back to a Message, decoding any bytes values."""
+        return cast(Message, decode_bytes_values(self.message))
+
+    @classmethod
+    def from_dict(cls, env: dict[str, Any]) -> "SessionMessage":
+        """Initialize a SessionMessage from a dictionary, ignoring keys that are not calss parameters."""
+        return cls(**{k: v for k, v in env.items() if k in inspect.signature(cls).parameters})
+
+
+@dataclass
+class SessionAgent:
```

Comment:
After reading through some of the code, `AgentSession` *might* read more clearly?  

I go back and forth on it. SessionMessage makes sense to me. However, when I read SessionAgent I think of an agent that specialized is Sessions whereas AgentSession is more clearly a session that belongs to an agent.